### PR TITLE
Feature/15 junit5 migration

### PR DESCRIPTION
**Pull request by Pyskaczu**

### CODE REVIEW

File: .idea/encodings.xml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (Pyskaczu) [inline .idea/encodings.xml:1]: add ".idea" to .gitignore

File: example/pom.xml [hunk 2]
```diff
@@ -63,46 +65,48 @@
...
+        </allure-cli-runner.version> <!-- Url to download Allure report CLI https://bintray.com/qameta/generic/allure2 -->
+
+        <aspectj.version>1.9.5</aspectj.version>
+        <allure-junit5.version>2.13.3</allure-junit5.version>
+        <allure-java-commons.version>2.13.3</allure-java-commons.version>
+        <allure-junit-platform.version>2.13.3</allure-junit-platform.version>
+        <allure-maven.version>2.10.0</allure-maven.version>
 
-        <!-- Cucumber version -->
+        <!-- Cucumber vallersion -->
```
Comment (Pyskaczu) [inline example/pom.xml:95]: Editorial error

File: example/pom.xml [hunk 3]
```diff
@@ -153,15 +157,37 @@
         </dependency>
 
         <!--This dependency is necessary for JUnit plugin. -->
+        <!--This dependency is necessary for JUnit plugin. -->
```
Comment (Pyskaczu) [inline example/pom.xml:160]: Doubled comment

File: example/pom.xml [hunk 4]
```diff
@@ -314,15 +367,14 @@
...
                             </properties>
                             <systemProperties>
-                                <property>
-                                    <name>allure.results.directory</name>
-                                    <value>${allure.report.directory}/allure-results</value>
-                                </property>
+                                <!--                                <property>-->
+                                <!--                                    <name>allure.results.directory</name>-->
+                                <!--                                    <value>${allure.report.directory}/allure-results</value>-->
+                                <!--                                </property>-->
```
Comment (Pyskaczu) [inline example/pom.xml:377]: To be removed

File: example/pom.xml [hunk 5]
```diff
@@ -336,12 +388,12 @@
...
                         <artifactId>allure-maven</artifactId>
-                        <version>2.9</version>
+                        <version>2.10.0</version>
                         <configuration>
                             <!-- Url to download Allure report CLI https://bintray.com/qameta/generic/allure2 -->
                             <reportVersion>${allure-cli-runner.version}</reportVersion>
-                            <resultsDirectory>${allure.report.directory}/allure-results</resultsDirectory>
-                            <reportDirectory>${allure.report.directory}/site/allure-report</reportDirectory>
+                            <!--                            <resultsDirectory>${allure.report.directory}/allure-results</resultsDirectory>-->
+                            <!--                            <reportDirectory>${allure.report.directory}/site/allure-report</reportDirectory>-->
```
Comment (Pyskaczu) [inline example/pom.xml:396]: To be removed

File: example/pom.xml [hunk 6]
```diff
@@ -526,24 +581,24 @@
...
                             </properties>
                             <systemProperties>
-                                <property>
-                                    <name>allure.results.directory</name>
-                                    <value>${allure.report.directory}/allure-results</value>
-                                </property>
+                                <!--                                <property>-->
+                                <!--                                    <name>allure.results.directory</name>-->
+                                <!--                                    <value>${allure.report.directory}/allure-results</value>-->
+                                <!--                                </property>-->
```
Comment (Pyskaczu) [inline example/pom.xml:592]: To be removed

File: example/pom.xml [hunk 7]
```diff
@@ -555,8 +610,8 @@
                         <configuration>
                             <!-- Url to download Allure report CLI https://bintray.com/qameta/generic/allure2 -->
                             <reportVersion>${allure-cli-runner.version}</reportVersion>
-                            <resultsDirectory>${allure.report.directory}/allure-results</resultsDirectory>
-                            <reportDirectory>${allure.report.directory}/site/allure-report</reportDirectory>
+                            <!--                            <resultsDirectory>${allure.report.directory}/allure-results</resultsDirectory>-->
+                            <!--                            <reportDirectory>${allure.report.directory}/site/allure-report</reportDirectory>-->
```
Comment (Pyskaczu) [inline example/pom.xml:614]: To be reomved

File: example/src/main/java/com/capgemini/mrchecker/webapi/BaseWebApiTest.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+	@Override
+	public void tearDown() {
+		
+	}
+	
+	@Override
+	public void setUp() {
+		
+	}
+}
```
Comment (Pyskaczu) [inline example/src/main/java/com/capgemini/mrchecker/webapi/BaseWebApiTest.java:15]: It looks this class is unnecessary.

File: example/src/resources/settings.properties [hunk 9]
```diff
@@ -6,7 +6,7 @@ selenium.phantomjs=./lib/webdrivers/phantomjs/bin/phantomjs.exe
 selenium.chrome=./lib/webdrivers/chrome/chromedriver.exe
 selenium.firefox=./lib/webdrivers/firefox/geckodriver.exe
 selenium.ie=./lib/webdrivers/internetexplorer/IEDriverServer.exe
-selenium.driverAutoUpdate=true
+selenium.driverAutoUpdate=false
```
Comment (Pyskaczu) [inline example/src/resources/settings.properties:9]: REvert

File: example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenExternalExampleTest.java [hunk 10]
```diff
@@ -1,72 +1,51 @@
...
+	
 	/**
 	 * ---------------------------------------------------------------------------------------------------------------
 	 * --- EXAMPLE 1 -------------------------------------------------------------------------------------------------
 	 * ---------------------------------------------------------------------------------------------------------------
 	 **/
-	@Test
-	@FileParameters("src/test/resources/datadriven/test.csv")
+	@ParameterizedTest
+	@CsvFileSource(resources = "/datadriven/test.csv", numLinesToSkip = 1)
```
Comment (Pyskaczu) [inline example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenExternalExampleTest.java:19]: Move  ", numLinesToSkip = 1" to Example2

File: example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenExternalExampleTest.java [hunk 11]
```diff
@@ -1,72 +1,51 @@
...
 	 * ---------------------------------------------------------------------------------------------------------------
 	 * --- EXAMPLE 3 -------------------------------------------------------------------------------------------------
 	 * ---------------------------------------------------------------------------------------------------------------
 	 * takes parameters from a CSV file and map them to specific mapper
 	 **/
-	@Test
-	@FileParameters(value = "src/test/resources/datadriven/test.csv", mapper = PersonMapper.class)
-	public void loadParamsFromAnyFile(Person person) {
+	@ParameterizedTest
+	@CsvFileSource(resources = "/datadriven/test.csv", numLinesToSkip = 1)
```
Comment (Pyskaczu) [inline example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenExternalExampleTest.java:46]: Remove ", numLinesToSkip..."

File: example/src/test/java/com/capgemini/mrchecker/cucumber/stepdefs/core/BaseStepDef.java [hunk 12]
```diff
@@ -1,10 +1,13 @@
...
 
 import com.capgemini.mrchecker.test.core.logger.BFLogger;
+
 import cucumber.api.java.en.Given;
+import io.qameta.allure.Step;
 
 public class BaseStepDef {
-
+	
+	@Step("asdad")
```
Comment (Pyskaczu) [inline example/src/test/java/com/capgemini/mrchecker/cucumber/stepdefs/core/BaseStepDef.java:10]: REmove

File: example/src/main/java/com/capgemini/mrchecker/Step.java [hunk 13]
```diff
@@ -0,0 +1,4 @@
+package com.capgemini.mrchecker;
+
+public @interface Step {
+}
```
Comment (Pyskaczu) [inline example/src/main/java/com/capgemini/mrchecker/Step.java:4]: To be removed

Comment (MalgorzataDzienia) [reply example/src/main/java/com/capgemini/mrchecker/Step.java:4]: Yes

**Comment by MalgorzataDzienia** [comment]
@Pyskaczu : I can see in repository two directories: example and template. Both with same tests examples. Which one is the right one and which one to use? Do we need both? Please help me to understand that.

**Comment by MajPat** [comment]
> 
> 
> @Pyskaczu : I can see in repository two directories: example and template. Both with same tests examples. Which one is the right one and which one to use? Do we need both? Please help me to understand that.
> 
> Maybe it is like that:
> first is filled with test examples
> second is almost empty so the end user doesn't have to delete test before stating own project.
> 
> Are both prepared for Junit5?

@MalgorzataDzienia,
Example contains tests that are using with various methods Mr.Checker.
Template is an almost empty project to use for new development.
We can use only the template project or use the parent project (main) to develop new tests in template and have the example tests in the repository.

File: example/src/main/java/com/capgemini/mrchecker/security/SecurityTest.java [hunk 14]
```diff
@@ -12,26 +14,16 @@
  * @author Marek Puchalski, Capgemini
  */
 public abstract class SecurityTest extends BaseTest {
-
-	private static final Injector        injector = Guice.createInjector(new SessionManagerModule());
-	private              ISessionManager sessionManager;
-
+	
+	private static final Injector	injector	= Guice.createInjector(new SessionManagerModule());
```
Comment (MajPat) [inline example/src/main/java/com/capgemini/mrchecker/security/SecurityTest.java:18]: Unused variable

**Comment by MalgorzataDzienia** [comment]
> Example contains tests that are using various methods from Mr.Checker.
> Template is an almost empty project to use for new development.
> We can use just the template project or use the parent project (main) to develop new tests in template and have the example tests in the repository.

Sure, but the template has to adapted for Junit5, too - min in pom file.

File: .idea/encodings.xml (hunk 1 above)
Comment (Pyskaczu) [reply .idea/encodings.xml:1]: Added

File: example/src/main/java/com/capgemini/mrchecker/webapi/BaseWebApiTest.java (hunk 8 above)
Comment (Pyskaczu) [reply example/src/main/java/com/capgemini/mrchecker/webapi/BaseWebApiTest.java:15]: Deleted

File: example/src/resources/settings.properties (hunk 9 above)
Comment (Pyskaczu) [reply example/src/resources/settings.properties:9]: reverted

File: example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenExternalExampleTest.java (hunk 11 above)
Comment (Pyskaczu) [reply example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenExternalExampleTest.java:46]: Removed

File: example/src/test/java/com/capgemini/mrchecker/cucumber/stepdefs/core/BaseStepDef.java (hunk 12 above)
Comment (Pyskaczu) [reply example/src/test/java/com/capgemini/mrchecker/cucumber/stepdefs/core/BaseStepDef.java:10]: removed

File: example/src/main/java/com/capgemini/mrchecker/Step.java (hunk 13 above)
Comment (Pyskaczu) [reply example/src/main/java/com/capgemini/mrchecker/Step.java:4]: removed

File: example/src/main/java/com/capgemini/mrchecker/security/SecurityTest.java (hunk 14 above)
Comment (Pyskaczu) [reply example/src/main/java/com/capgemini/mrchecker/security/SecurityTest.java:18]: Reverted to use Guice again.

File: example/pom.xml (hunk 7 above)
Comment (Pyskaczu) [reply example/pom.xml:614]: removed

**Comment by Pyskaczu** [comment]
@MalgorzataDzienia Right, the template folder needs adjustments. I'm on it.

File: example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenInternalExampleTest.java [hunk 15]
```diff
@@ -1,96 +1,96 @@
...
+	// @ParameterizedTest
+	// @ValueSource(ints ={1, 2, 3},
+	// {3, 4, 7},
+	// {5, 6, 11},
+	// {{7, 8, 15}})
+	// public void addProducesCorrectValue_usingAnnotatedValueSource(final int a, final int b, final int expectedResult)
+	// {
+	// assertEquals(expectedResult, testSubject.add(a, b));
+	// }
+
```
Comment (MajPat) [inline example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenInternalExampleTest.java:35]: I think we should get rid of commented and unsupported tests

Comment (Pyskaczu) [reply example/src/test/java/com/capgemini/mrchecker/core/datadriven/DataDrivenInternalExampleTest.java]: Deleted

**Pull request merged**